### DISCUSSION TITLE
Output files when MismatchCorrector is enabled

### DISCUSSION THREAD

**Discussion by sallesmath** [ablab/spades#1003] [Q&A]
### Question

Hi!

I just ran SPAdes with the MismatchCorrector option enabled (--carefull option).

And now I have three files with assembled contigs. Could you tell me if they are the same, or do they correspond to different assemblies?

These are the files I'm talking about:
1) 'contigs.fasta' file, located on the first page of the results folder
2) 'assembled_contigs' file, located in the 'misc' folder
3) 'corrected_contigs' file, located in the /mismatch_corrector/contigs folder

Thanks!

### spades.log

[spades.log](https://github.com/ablab/spades/files/9115246/spades.log)

### params.txt

[params.txt](https://github.com/ablab/spades/files/9115259/params.txt)

### SPAdes version

SPAdes 3.15.4

### Operating System

Linux-5.4.72-microsoft-standard-WSL2-x86_64

### Python Version

2.7.16

### Method of SPAdes installation

manual

### No errors reported in spades.log

- [X] Yes

**Comment by asl** [comment]
Hello

The log clearly reads:
```
 * Corrected reads are in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/corrected/
 * Assembled contigs are in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/contigs.fasta
 * Assembled scaffolds are in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/scaffolds.fasta
 * Paths in the assembly graph corresponding to the contigs are in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/contigs.paths
 * Paths in the assembly graph corresponding to the scaffolds are in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/scaffolds.paths
 * Assembly graph is in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/assembly_graph.fastg
 * Assembly graph in GFA format is in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/assembly_graph_with_scaffolds.gfa
```

Hope this helps. SPAdes manual (https://cab.spbu.ru/files/release3.15.4/manual.html#spadesoutput) contains more information about SPAdes output

**Comment by sallesmath** [comment]
I apologize, but I'm still confused. As a post processing tool, MismatchCorrector is not acting on the resulting contigs?

That 'new_test_amcc204364_spades/corrected/' folder only contains corrected **reads**, not contigs.

Anyway, is there no difference between the files I mentioned earlier? See below:

1) 'contigs.fasta' file, located on the first page of the results folder
2) 'assembled_contigs.fasta' file, located in the 'misc' folder
3) 'corrected_contigs.fasta' file, located in the /mismatch_corrector/contigs folder

My question is whether these three files have the same content or not. I didn't find this information in the manual.

Thanks again!

**Reply by asl** [reply]
Your output is as follows:

```
* Assembled contigs are in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/contigs.fasta
* Assembled scaffolds are in /home/sallesmath/dados_uce/phyluce_dados_uce_tropidurus/dados_andre/spades-assemblies/new_test_amcc204364_spades/scaffolds.fasta
```

Everything else are internal assembler files and not intended for any external usage.

**Reply by sallesmath** [reply]
Got it.

Thanks a lot.
